astropy/io/fits/hdu/table.py

```diff
@@ -616,7 +616,7 @@ def _update_column_attribute_changed(self, column, col_idx, attr,
             else:
```

Comment:
Ah, thanks for the explanation, @MSeifert04 ! Okay, along that line, I can get the code block to run as follows, @astrofrog .

```python
import numpy as np
from astropy.io import fits

x = np.array([1,2,3])
col = fits.Column(name='a', array=x, format='E')
hdu = fits.BinTableHDU.from_columns([col])
del hdu._header['TTYPE1']
hdu.columns[0].name = 'b'
```